presto-main/src/main/java/com/facebook/presto/metadata/FunctionMetadata.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.metadata;
+
+import com.facebook.presto.spi.function.FunctionKind;
+import com.facebook.presto.spi.function.Signature;
+import com.facebook.presto.spi.type.TypeSignature;
+
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public class FunctionMetadata
+{
+    private final Signature signature;
```

Comment:
Why not using `FunctionHandle`?

Reply:
You use a `FunctionHandle` to fetch `FunctionMetadata` `functionManager.getFunctionMetadata(FunctionHandle)`. It would be really strange if metadata is a wrapper on handle... What's the point... The implementation of this will likely change in the future, as I mentioned in https://github.com/prestodb/presto/pull/12570#discussion_r271081765. But it's not likely to ever hold a `FunctionHandle` as a member. 😛